src/artifact.cpp

```diff
@@ -1260,7 +1260,7 @@ void it_artifact_tool::serialize(JsonOut &json) const
     }
     json.end_array();
     json.member("volume", volume / units::legacy_volume_factor);
-    json.member("weight", weight);
+    json.member("weight", to_gram( weight ) );
```

Comment:
No, it's the magic known as [ADL](http://en.cppreference.com/w/cpp/language/adl). The type of `weight` is defined in the `units` namespace, so `to_gram` is searched in that namespace as well.